users/userRouter.js

```diff
@@ -1,47 +1,154 @@
-const express = 'express';
+const express = require('express');
 
-const router = express.Router();
+const userDb = require('./userDb.js');
+const postDb = require('../posts/postDb');
 
-router.post('/', (req, res) => {
+const router = express.Router();
 
+router.post('/', validateUser, (req, res) => {
+  const user = req.body;
+  userDb
+    .insert(user)
+    .then(user => {
+      res.status(201).json(user);
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: 'There was an error saving the user to the database' });
+    });
 });
 
-router.post('/:id/posts', (req, res) => {
-
+router.post('/:id/posts', validateUserId, validatePost, (req, res) => {
+  let post = req.body;
+  const id = req.params.id;
+  post.user_id = id;
+  console.log(req.user);
+
+  // if (!post.text) {
+  //   res.status(404).json({
+  //     message: 'Please provide text for the post',
+  //   });
+  // } else {
+  postDb
+    .insert(post)
+    .then(post => {
+      res.status(201).json(post);
+    })
+    .catch(err => {
+      res.status(500).json({
+        error: 'There was an error while saving the post to the database',
+      });
+    });
+  // }
 });
 
 router.get('/', (req, res) => {
-
+  userDb
+    .get()
+    .then(users => {
+      res.status(200).json(users);
+    })
+    .catch(err => {
+      res.status(500).json({ error: 'There was an error retrieving users' });
+    });
 });
 
-router.get('/:id', (req, res) => {
-
+router.get('/:id', validateUserId, (req, res) => {
+  userDb
+    .getById(req.params.id)
+    .then(user => {
+      res.status(200).json(user);
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: 'A user with that specified ID does not exist' });
+    });
 });
 
-router.get('/:id/posts', (req, res) => {
-
+router.get('/:id/posts', validateUserId, (req, res) => {
+  userDb
+    .getUserPosts(req.params.id)
+    .then(posts => {
+      if (posts.length > 0) {
```

Comment:
I'm getting extra nit-picky today: `!!posts.length` turns this into an actual boolean. But, this conditional shouldn't be here at all, it would be appropriate to return an empty array for a user who has no posts.